### PR TITLE
add KairosDB registry

### PR DESCRIPTION
**Pull request by ilinchik**
Adds a new meter registry which enables support for KairosDB (#761)

### CODE REVIEW

File: implementations/micrometer-registry-kairos/src/main/java/io/micrometer/kairos/KairosMeterRegistry.java [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+            if (StringUtils.isNotBlank(config.userName()) && StringUtils.isNotBlank(config.password())) {
+                byte[] authBinary = (config.userName() + ":" + config.password()).getBytes(StandardCharsets.UTF_8);
+                String authEncoded = Base64.getEncoder().encodeToString(authBinary);
+                con.setRequestProperty(HttpHeader.AUTHORIZATION, "Basic " + authEncoded);
+            }
+            logger.trace("Sending payload to KairosDB:");
+            logger.trace(bulkPayload);
+
+            try (OutputStream os = con.getOutputStream()) {
+                os.write(bulkPayload.getBytes());
```
Comment (pjfanning) [inline implementations/micrometer-registry-kairos/src/main/java/io/micrometer/kairos/KairosMeterRegistry.java]: could you use getBytes("utf-8") to ensure the char encoding is correct?

File: implementations/micrometer-registry-kairos/src/main/java/io/micrometer/kairos/KairosMeterRegistry.java [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+    }
+
+    private static class KairosMetricBuilder {
+        private StringBuilder sb = new StringBuilder("{");
+
+        KairosMetricBuilder field(String key, String value) {
+            if (sb.length() > 1) {
+                sb.append(",");
+            }
+            sb.append("\"").append(key).append("\":\"").append(value).append("\"");
```
Comment (pjfanning) [inline implementations/micrometer-registry-kairos/src/main/java/io/micrometer/kairos/KairosMeterRegistry.java]: append('\"') can be used to append single characters

File: implementations/micrometer-registry-kairos/src/main/java/io/micrometer/kairos/KairosConfig.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
+package io.micrometer.kairos;
```
Comment (pjfanning) [inline implementations/micrometer-registry-kairos/src/main/java/io/micrometer/kairos/KairosConfig.java]: could you add the Pivotal license to all source files?

File: implementations/micrometer-registry-kairos/src/main/java/io/micrometer/kairos/KairosMeterRegistry.java (hunk 1 above)
Comment (ilinchik) [reply implementations/micrometer-registry-kairos/src/main/java/io/micrometer/kairos/KairosMeterRegistry.java]: yes sure, fixed

File: implementations/micrometer-registry-kairos/src/main/java/io/micrometer/kairos/KairosMeterRegistry.java (hunk 2 above)
Comment (ilinchik) [reply implementations/micrometer-registry-kairos/src/main/java/io/micrometer/kairos/KairosMeterRegistry.java]: thanks, fixed

File: implementations/micrometer-registry-kairos/src/main/java/io/micrometer/kairos/KairosConfig.java (hunk 3 above)
Comment (ilinchik) [reply implementations/micrometer-registry-kairos/src/main/java/io/micrometer/kairos/KairosConfig.java]: done

**Comment by marbon87** [comment]
any update for this PR?

**Comment by jkschneider** [comment]
Merged after bringing it up to the latest changes in core with 6ad61a24.

**Pull request closed without merging**